### DISCUSSION TITLE
Noob Question: Debugging & Updating

### DISCUSSION THREAD

**Discussion by xsova** [dustinlyons/nixos-config#117] [Q&A]
1. Is there a way to debug the build-switch exec? I know there exists a `nix flake check /path/to/flake` but I'm not sure that the same thing exists for `nix run`. I've been trying to switch all my configs over to this, and it seems like a bit of a pain point for me when I am exiting my text editor (helix).. Maybe there's a debugger configuration? 

2. How would I go about updating _everything_ like home-manager, nix, etc? I see in hosts/darwin/default.nix the system.stateVersion is 4, and the (I think?) current version is 24.05. Is there a particular reason that one would want to avoid updating it? What would happen if I did change that value and reran build-switch?

**Comment by jplee-metaframe** [comment]
RE Updating: I'm just doing:
```
nix flake update
```
Seems to work.
